Libraries/Components/RefreshControl/RefreshControl.js

```diff
@@ -124,7 +124,22 @@ const RefreshControl = React.createClass({
     size: React.PropTypes.oneOf(RefreshLayoutConsts.SIZE.DEFAULT, RefreshLayoutConsts.SIZE.LARGE),
   },
 
-  _nativeRef: {},
+  _nativeRef: (null: any),
+  _lastNativeRefreshing: false,
+
+  componentDidMount() {
+    this._lastNativeRefreshing = this.props.refreshing;
+  },
+
+  componentWillReceiveProps(nextProps: {refreshing: boolean}) {
```

Comment:
I'd like to keep componentWillReceiveProps pure (with the exception of setting `state`) whenever possible. Can we instead do:

```
componentDidUpdate() {
  if (this.props.refreshing !== nextProps.refreshing) {
    // refreshing was updated via the normal rerender
    this._lastNativeRefreshing = this.props.refreshing;
  } else if (this.props.refreshing !== this._lastNativeRefreshing) {
    // setNativeProps
  }
}
```

though personally I find it easier to reason about to cache and always pass the same value to `refreshing` in the render function (as I did in PickerAndroid) so that you know exactly when the native `refreshing` value gets updated.